### ISSUE TITLE
can't use GPTSimpleVectorIndex with Microsoft Azure OpenAI

### ISSUE THREAD

**Issue by alexeenka** [jerryjliu/gpt_index#332]
I can't use GPTSimpleVectorIndex with Microsoft Azure OpenAI

I have the following settings:

```
openai.api_type = "azure"
openai.api_base = "https://xyz.openai.azure.com/"
openai.api_version = "2022-12-01"
openai.api_key = os.getenv("OPENAI_API_KEY")

from gpt_index import GPTSimpleVectorIndex, SimpleDirectoryReader, LLMPredictor

documents = SimpleDirectoryReader('docs/test-data', recursive=True, required_exts=[".md"]).load_data()
llm_predictor = LLMPredictor(llm=AzureOpenAI(deployment_name="xyz-Curie", openai_api_key=openai.api_key))
index = GPTSimpleVectorIndex(documents=documents, llm_predictor=llm_predictor)

```

Error: 

`openai.error.InvalidRequestError: The API deployment for the resource does not exist.`

**Comment by jerryjliu** [comment]
i haven't really tested the AzureOpenAI stuff, i think we only have embedding support atm and not the full llm? https://github.com/jerryjliu/gpt_index/pull/254/files

**Comment by hophanms** [comment]
Any update on this? Not sure where to get `AzureOpenAI`

**Comment by schthms** [comment]
It works if you specify both embedding and completion model:
```
embed_model = OpenAIEmbedding(deployment_name="text-search-ada-doc-001")
llm_predictor = LLMPredictor(llm=AzureOpenAI(deployment_name="davinci-003", openai_api_key=openai.api_key))
index = GPTSimpleVectorIndex(documents, embed_model=embed_model, llm_predictor=llm_predictor)
```

**Comment by jerryjliu** [comment]
should be resolved with the nb provided in #402
